codegen/src/useCli.ts

```diff
@@ -0,0 +1,34 @@
+import { spawn } from 'child_process';
+import { Ora } from 'ora';
+
+import { cliPath } from './cliPath';
+
+export const useCli = ({ spinner }: { spinner: Ora }) =>
+  new Promise<void>((resolve, reject) => {
+    spinner.info('Delegating to Xata CLI...');
+
+    const cli = spawn(cliPath, ['init'], { stdio: ['inherit', 'pipe', 'pipe'] })
+      .on('close', (code) => {
+        if (code !== 0) {
+          return;
+        }
+        resolve();
+      })
+      .on('error', (e: unknown) => {
+        reject(e);
+      });
+
+    cli.stdout.pipe(process.stdout);
+    cli.stdout.on('data', async (c) => {
+      const line = c.toString();
+      if (line !== 'Xata CLI is not configured, please run `xata auth login`\n') {
```

Comment:
This would break if the xata CLI changes the message. Suggestions:
- Use a regex to be more flexible and accept small changes in the message.
- Make the xata cli return a specific error code and use that in the `close` event.